docs/2-building-blocks-of-oop-part-1/hometask/src/Triangle.ts

```diff
@@ -1,3 +1,56 @@
 import { Shape } from './Shape';
+import { Point } from './Point';
 
-export class Triangle extends Shape {}
+export class Triangle extends Shape {
+  constructor(point1: Point, point2: Point, point3: Point);
+  constructor(
+    point1: Point,
+    point2: Point,
+    point3: Point,
+    color: string,
+    filled: boolean
+  );
+  constructor(
+    point1: Point,
+    point2: Point,
+    point3: Point,
+    color?: string,
+    filled?: boolean
+  ) {
+    if (typeof color === 'string' && typeof filled === 'boolean') {
+      super([point1, point2, point3], color, filled);
+    } else {
+      super([point1, point2, point3]);
+    }
+  }
+
+  toString() {
+    return `Triangle[${this.points
+      .map((point, index) => `v${index + 1}=${point.toString()}`)
+      .join(',')}]`;
+  }
+
+  getType(): string {
+    const distances = this.points.map((point, index, points) => {
+      let distance: number;
+      if (index === 0) {
+        distance = point.distance(points[points.length - 1]);
+      } else {
+        distance = point.distance(points[index - 1]);
+      }
+      return parseFloat(distance.toFixed(2));
+    });
+
+    const equalSides = distances.filter(
```

Comment:
Could you pls check https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set and try to use `Set.size` to simplify this logic?

Reply:
thank you, will have a look and try to implement.